.github/workflows/ci.yml

```
@@ -0,0 +1,68 @@
+    - name: Run Pylint
+      run: |
+        pylint src/ --output-format=colorized --reports=no --score=no --rcfile=pyproject.toml
+
+    - name: Run tests (if present)
+      run: |
+        if [ -d "tests" ] || find . -name "test_*.py" -o -name "*_test.py" | grep -q .; then
+          poetry run pytest -v --tb=short
+        else
+          echo "No tests found, skipping test step"
+        fi
+      continue-on-error: true 
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)